src/RegisteredDomain.php

```
@@ -119,9 +119,18 @@ private static function normalizeHost(string $input): string
         }
         $host = trim(mb_strtolower($host, 'UTF-8'));
         if ($host !== '' && $host[0] === '[') {
-            $host = trim($host, '[]');
+            // IPv6 literal, e.g. [::1] or [::1]:443
+            // Extract the address between brackets, discarding any trailing port.
+            if (preg_match('/^\[([^\]]+)]/', $host, $m)) {
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $m. Configured minimum length is 3. (undefined)

(ShortVariable)